spec/support/fake_tmpdir_helper.rb

```diff
@@ -0,0 +1,227 @@
+# = fake_tmpdir_helper.rb
+#
+# Spec::Support::FakeTmpdirHelper is a helper module that is used to simulate a
+# logical volume `/tmp` directory with limited disk space on the file system.
+#
+# To do this, it sets up a loopback device, which is just a zero'd out file
+# (defaults to `manageiq/tmp/fake_fs.img`), and mounts it.  `Dir.tmpdir` is
+# then overwritten to point to that newly mounted device, and it can
+# transparently be used to be tested against using file system commands like
+# normal (`df -P /fake_fs` for example).
+#
+# == Usage
+#
+# To use this lib, include it inside of your `describe` block:
+#
+#     describe 'Something' do
+#       include Spec::Support::FakeTmpdirHelper
+#     end
+#
+# Then use it by calling `with_fake_tmp` inside of the spec where you want to
+# have a "fake tmp dir".  It will do all of the setup and cleanup within the
+# block.
+#
+# Note:  This class is most likely **NOT** thread safe currently
+
+require 'tmpdir'
+require 'tempfile'
+require 'pathname'
+require 'sys-uname'
+require 'active_support/all'
+
+module Spec
+  module Support
+    module FakeTmpdirHelper
+      PROJECT_TEMP_DIR = Pathname.new(File.expand_path("../../tmp", __dir__))
+
+      module_function
+
+      # The main function for setting up a loopback device.
+      #
+      #     with_fake_tmp do
+      #       expect(Dir.tmpdir).to eq(fake_tmp_dir)
+      #     end
+      #
+      def with_fake_tmp(size = fake_fs_size)
+        build_file(size)
+        attach_tmp_fs
+        patch_tmpdir
+
+        yield
+      ensure
+        detach_fake_fs
+        unpatch_tmpdir
+      end
+
+      # rubocop:disable Lint/DuplicateMethods
+      case Sys::Platform::IMPL
+      when :linux
+        # Linux:  Sets up the loop back device, and mounts it.
+        #
+        # Will effectively run the following:
+        #
+        #     $ losetup -f
+        #     #=> /dev/loop0
+        #     $ losetup /dev/loop0 tmp/fake_fs.img
+        #     $ mkfs.ext4 /dev/loop0
+        #     $ mount /dev/loop0 tmp/fake_fs
+        #     $ chmod -R 777 tmp/fake_fs
+        #
+        # Note:  The `chmod` call is required since we are setting this up with
+        # `sudo`, so we need to give write permissions to the world anything
+        # within this dir so the `rspec` process can still write to it in the
+        # specs.
+        #
+        def attach_tmp_fs
+          @dev_disk = `losetup -f`.chomp.strip
+          cmds = [
+            "sudo losetup #{@dev_disk} #{tmp_fs_file}",
```

Comment:
Does this work running locally on a linux workstation? I'd imagine the suite would halt if you didn't run it as root, right?

Reply:
Just to answer this:

The assumption (which is with `sudo` on Vagrant on Travis, and my two places I tested) is that using it with those users doesn't require entering a password.  So to your question, I assume if `sudo` does require a password on your machine, then yes, the scenario you described probably would happen.

So another good reason to can this stuff for now 😄